### PR TITLE
HPCC-17725 Add part has extra fields

### PR DESCRIPTION
**Pull request by miguelvazq**
The fields "Target", "Process", and "packagemap" should always come from the parent tabs... not overridden by user selection. Now we are sending the values in the request and not displaying since it is not editable.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [x] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browers
  - [x] The component(s) render as expected

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review.

File: esp/src/eclwatch/PackageMapPartsWidget.js [hunk 1]
```diff
@@ -93,9 +93,8 @@ define([
                     var addPartInput = context.getFilter();
                     WsPackageMaps.AddPartToPackageMap({
                         request: {
-                            Target: addPartInput.AddPartsTarget,
-                            Process: addPartInput.Process,
-                            PackageMap: addPartInput.PackageMap,
+                            Target: context.params.target,
+                            PackageMap: context.params.packageMap.split('::')[1],
```
Comment (GordonSmith) [inline esp/src/eclwatch/PackageMapPartsWidget.js]: Do ew know that there will always be at least one instance of "::"?

**Review by GordonSmith — APPROVED** [review]
As long as we know "::" will always exist in the package name then this is ok.

File: esp/src/eclwatch/PackageMapPartsWidget.js (hunk 1 above)
Comment (afishbeck) [inline esp/src/eclwatch/PackageMapPartsWidget.js]: the TARGET:: part of the PackageMap id is prepended internally.  There is support for global (unscopped) packagemap ids, although rarely used you should probably support IDs that don't start with TARGET::.

Comment (miguelvazq) [inline esp/src/eclwatch/PackageMapPartsWidget.js]: @afishbeck Can you verify? The package map from the previous screen is always separated by '::' so I am trimming to pass it in the request.

Comment (afishbeck) [inline esp/src/eclwatch/PackageMapPartsWidget.js]: @miguelvazq to clarify the id will usually have the target prepended... but there is a way to create one from command line that doesn't have that... so you should probably trim it off if its there, but use the whole string if it's not there.

**Comment by miguelvazq** [comment]
@GordonSmith Added support for those containing '::' and those not containing

File: esp/src/eclwatch/PackageMapPartsWidget.js [hunk 2]
```diff
@@ -91,11 +91,15 @@ define([
              this.addPartsDropDown.on("apply", function (evt) {
                 if (context.addPartsDropDown.filterForm.validate()){
                     var addPartInput = context.getFilter();
+                    var packageMapSearch = context.params.packageMap.search("::");
+                    var packageMapClean;
+
+                    packageMapSearch > -1 ? packageMapClean = context.params.packageMap.split('::')[1] : packageMapClean = context.params.packageMap;
+
```
Comment (afishbeck) [inline esp/src/eclwatch/PackageMapPartsWidget.js]: looks good

Comment (GordonSmith) [inline esp/src/eclwatch/PackageMapPartsWidget.js]: Looks ok (Assuming there is at most 1 instance of "::"

Comment (afishbeck) [inline esp/src/eclwatch/PackageMapPartsWidget.js]: @GordonSmith I only add one, and it's always in the form of "<target>::".  People usually use the file name for the rest of the ID.  I probably SHOULD have disallowed "::" in the ID itself, but as it is I suppose someone could override the name to use :: as part of the Id. 

You know the target, so the solution that is probably most correct is to look for a leading "<target>::" and remove it if its there.  Probably do case insensitive compare just to be safe.

Comment (GordonSmith) [inline esp/src/eclwatch/PackageMapPartsWidget.js]: @miguelvazq @afishbeck - is this ok to pull as is?

Comment (afishbeck) [inline esp/src/eclwatch/PackageMapPartsWidget.js]: I don't know why anyone would want to do it, but I was able to add a packagefile with :: in the name... which meant roxie::xxx::yyy.pkg was created, and there is also a way of setting the name explicitly to whatever you want.

The only way to support such names though is to specifically remove <target>:: from the beginning.  If the consensus is that this should be a known limitation so be it.

Comment (richardkchapman) [inline esp/src/eclwatch/PackageMapPartsWidget.js]: I'm inclined to merge as is and open a new Jira to address that once we have worked out how we want to do so. 

Any objections?

Comment (afishbeck) [inline esp/src/eclwatch/PackageMapPartsWidget.js]: Fine with me.

Comment (miguelvazq) [inline esp/src/eclwatch/PackageMapPartsWidget.js]: I'm good with that.

**Comment by miguelvazq** [comment]
@g-pan FYI the screen has changed for the add part.

**Pull request merged**